Example/Okta/ViewController.swift

```diff
@@ -66,9 +67,17 @@ class ViewController: UIViewController {
     }
 
     func loginCodeFlow() {
-        OktaAuth.login().start(self)
-        .then { _ in self.buildTokenTextView() }
-        .catch { error in print(error) }
+        let promise: Promise<OktaTokenManager>
+        if ProcessInfo.processInfo.environment["UITEST"] == "1" {
+            let config = ["issuer": ProcessInfo.processInfo.environment["ISSUER"]!,
+                          "clientId": ProcessInfo.processInfo.environment["CLIENT_ID"]!,
+                          "redirectUri": ProcessInfo.processInfo.environment["REDIR_URI"]!,
+                          "scopes": "openid profile offline_access"]
+            promise = OktaAuth.login().start(withDictConfig: config, view: self)
+        } else {
+            promise = OktaAuth.login().start(self)
+        }
+        promise.then { _ in self.buildTokenTextView() }.catch { error in print(error) }
```

Comment:
I know that this approach removes the need to handle success and error states for both, but we probably do not need to expose the inner Promise mechanics to users of the example app.

Although it is **more** code, what do you think about something like:

```swift
var environmentConfig: [String: Any]?

if ProcessInfo.processInfo.environment["UITEST"] == "1" {
  environmentConfig = /* set the config */
}

if let config = environmentConfig {
  OktaAuth.login().start(withDictConfig: config, view: self)
    .then { _ in self.buildTokenTextView() }.catch { error in print(error) }
} else {
  OktaAuth.login().start(self)
    .then { _ in self.buildTokenTextView() }.catch { error in print(error) }
}


```

Reply:
changed